src/java.base/share/classes/java/lang/invoke/MethodHandleStatics.java

```diff
@@ -88,6 +89,7 @@ private MethodHandleStatics() { }  // do not instantiate
                 props.getProperty("java.lang.invoke.MethodHandle.PROFILE_GWT", "true"));
         CUSTOMIZE_THRESHOLD = Integer.parseInt(
                 props.getProperty("java.lang.invoke.MethodHandle.CUSTOMIZE_THRESHOLD", "127"));
+        NO_SOFT_CACHE = CDS.disableMethodHandleSoftCache();
```

Comment:
I suggest to use `CDS.disableMethodHandleSoftCache()` as the default and allow to override the mode through a property.

```
NO_SOFT_CACHE = Boolean.parseBoolean( 
        props.getProperty("java.lang.invoke.MethodHandle.NO_SOFT_CACHE", 
                                        Boolean.toString(CDS.disableMethodHandleSoftCache()));
```

Reply:
My initial implementation used a `java.lang.invoke.MethodHandle.NO_SOFT_CACHE` property. However, I was afraid that such a property might be abused, which would cause problems with class unloading. Therefore, I decided to make it fully controlled by HotSpot via a native method.

Reply:
After further offline discussion with @iwanowww, I have reverted back to the system property as suggested.